pkg/core/resources/apis/mesh/validators.go

```diff
@@ -150,3 +151,22 @@ func ProtocolValidator(protocols ...string) SelectorValidatorFunc {
 		return
 	}
 }
+
+var nameCharacterSet = regexp.MustCompile(`^[a-zA-Z0-9\.\-_:]*$`)
+var selectorCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*|\*)$`)
+
+func SelectorCharacterSetValidator(path validators.PathBuilder, selector map[string]string) validators.ValidationError {
```

Comment:
To be honest, I'd use this in `ValidateSelector` so it's checked on every policy, not only on fault injection